gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -69,6 +72,209 @@ String entry() {
     }
   }
 
+  public static abstract class EntityField {
+
+    private final String selector;
+
+    public EntityField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+  }
+
+  public static class AclField extends EntityField {
+
+    public AclField(String selector) {
+      super(selector);
+    }
+
+    public static AclField bucket() {
+      return new AclField("bucket");
+    }
+
+    public static AclField domain() {
+      return new AclField("domain");
+    }
+
+    public static AclField email() {
+      return new AclField("email");
+    }
+
+    public static AclField entity() {
+      return new AclField("entity");
+    }
+
+    public static AclField entityId() {
+      return new AclField("entityId");
+    }
+
+    public static AclField etag() {
+      return new AclField("etag");
+    }
+
+    public static AclField generation() {
+      return new AclField("generation");
+    }
+
+    public static AclField id() {
+      return new AclField("id");
+    }
+
+    public static AclField kind() {
+      return new AclField("kind");
+    }
+
+    public static AclField object() {
+      return new AclField("object");
+    }
+
+    public static AclField projectTeam() {
+      return new AclField("projectTeam");
+    }
+
+    public static AclField role() {
+      return new AclField("role");
+    }
+
+    public static AclField selfLink() {
+      return new AclField("selfLink");
+    }
+
+    static String selector(AclField... aclFields) {
+      HashSet<String> fieldStrings = Sets.newHashSetWithExpectedSize(aclFields.length + 2);
+      fieldStrings.add(AclField.role().selector());
+      fieldStrings.add(AclField.entity().selector());
+      for (AclField field : aclFields) {
+        fieldStrings.add(field.selector());
+      }
+      return new StringBuffer()
+          .append("acl(")
+          .append(Joiner.on(",").join(fieldStrings))
+          .append(")")
+          .toString();
+    }
+  }
+
+  public static class BlobField extends EntityField {
+
+    public BlobField(String selector) {
+      super(selector);
+    }
+
+    public static BlobField acl(AclField... aclFields) {
```

Comment:
Did you consider just having an "acl" field in this level (basically allowing with or without acl but not getting part of an ACL)? I do wonder how partial ACL will work with patch (with or without id).